File: 0257. 二叉树的所有路径/DFS.js
// 84ms
var binaryTreePaths = function(root) {
    var current = [], result = []
    function search(node) {
        current.push(node.val)
        if (node.left == null && node.right == null) {
            result.push(current.join("->"))
        } else {
            if (node.left != null) {
                search(node.left)
            }
            if (node.right != null) {
                search(node.right)
            }
        }
        current.pop()
    }
    if (root != null) {
        search(root)
    }
    return result
};

File: 0144. 二叉树的前序遍历/DFS.swift
// 4ms
class Solution {
    func preorderTraversal(_ root: TreeNode?) -> [Int] {
        var result = [Int]()
        func preOrder(_ root: TreeNode?) {
            if let root = root {
                result.append(root.val)
                preOrder(root.left)
                preOrder(root.right)
            }
        }
        preOrder(root)
        return result
    }
}